src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+	 */
+	public function get_collection_params() {
+		$params                       = parent::get_collection_params();
+		$params['context']['default'] = 'view';
+
+		if ( isset( $params['per_page']['default'] ) ) {
+			$params['per_page']['default'] = 5;
+		}
+
+		$params['user_id'] = array(
+			'description'       => __( 'Limit result set to items concerning a specific user (ID).', 'buddypress' ),
+			'default'           => 0,
+			'type'              => 'integer',
+			'sanitize_callback' => 'absint',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['status'] = array(
+			'description'       => __( 'Limit result set to items with a specific status.', 'buddypress' ),
+			'default'           => 'unread',
+			'type'              => 'string',
+			'items'             => array(
+				'enum' => array( 'unread', 'dismissed' ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => 'sanitize_key',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['type'] = array(
+			'description'       => __( 'Limit result set to items with a specific type.', 'buddypress' ),
+			'default'           => 'active',
+			'type'              => 'string',
+			'items'             => array(
+				'enum' => array( 'active', 'inactive' ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => 'sanitize_key',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['target'] = array(
+			'description'       => __( 'Limit result set to items concerning one or more specific targets.', 'buddypress' ),
+			'type'              => 'array',
+			'items'             => array(
+				'enum' => array( 'community', 'contributors', 'admins' ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => 'wp_parse_list',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['priority'] = array(
+			'description'       => __( 'Limit result set to items having a specific priority.', 'buddypress' ),
+			'type'              => 'integer',
+			'items'             => array(
+				'enum' => array( 1, 2, 3 ),
+				'type' => 'integer',
+			),
+			'sanitize_callback' => 'absint',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		/**
+		 * Filters the collection query params.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param array $params Query params.
+		 */
+		return apply_filters( 'bp_members_notices_rest_collection_params', $params );
+	}
+
+	/**
+	 * Prepare a notice for creation or update.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return BP_Members_Notice|WP_Error Object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$schema        = $this->get_item_schema();
+		$notice_id     = $request->get_param( 'id' );
+		$prepared_item = bp_members_get_notice( $notice_id );
+
+		// Notice ID.
+		if ( ! empty( $schema['properties']['id'] ) && ! empty( $prepared_item->id ) ) {
+			$prepared_item->id = $prepared_item->id;
+		}
```

Comment:
Oddly, if the `id` property is empty, we are passing it along anyway. So I'm not sure how useful this conditional actually is.

Reply:
I've edited it to remove the ID if the schema doesn't have the ID property, I guess it's to prepare a create request.